rust/src/domains/polars/series/mod.rs

```diff
@@ -215,12 +215,64 @@ impl<T: CheckAtom + PrimitiveDataType> SeriesAtomDomain for OptionDomain<AtomDom
     const NULLABLE: bool = true;
 }
 
+#[derive(Debug, Clone, PartialEq, Default)]
+pub struct CategoricalDomain {
+    /// The encoding used to assign numerical indices to each known possible category.
+    encoding: Option<PlIndexSet<String>>,
+}
+
+impl CategoricalDomain {
+    /// Only use this constructor if you know both the category set,
+    /// as well as how the categories are encoded.
+    ///
+    /// Typically when Polars constructs categorical data,
+    /// it assigns indices by the order encountered in the data, making the encoding data-dependent.
+    /// An example where this can be called is for categorical data emitted by the Polars cut expression,
+    /// where the categories and encoding are pre-determined by the expression (the bin edges).
```

Comment:
This is interesting. I had tried to use category at some point in a docs example, for something like state abbreviations, but it sounds like we would never recommend that. (But this isn't bootstrapped, so it's not showing up in the generated code in any case.)

Maybe the actionable thing here is to note in our Polars table that categories aren't available, and explain why not? File an issue from this comment if that seems like a good route.

Reply:
I've gone ahead and fleshed this PR out to actually add categoricals. You can use categorical data with unknown encodings, so long as you don't read or change the encoding.